clients/src/main/java/org/apache/kafka/clients/consumer/Consumer.java

```diff
@@ -71,12 +71,12 @@
     /**
      * @see KafkaConsumer#commit(Map, CommitType)
      */
-    public void commit(Map<TopicPartition, Long> offsets, CommitType commitType);
+    public void commit(Map<TopicPartition, OffsetMetadata> offsets, CommitType commitType);
```

Comment:
@ijuma I personally don't think it's too bad, but it does add some semantic overhead to the API. Most users will probably just want to commit offsets, but this forces them to deal with the notion of associated metadata (even if they can just leave it empty).